modules/core/src/core/animation-loop.js

```diff
@@ -98,41 +103,46 @@ export default class AnimationLoop {
   // Starts a render loop if not already running
   // @param {Object} context - contains frame specific info (E.g. tick, width, height, etc)
   start(opts = {}) {
-    this._stopped = false;
+    if (this._running) {
+      return this;
+    }
+    this._running = true;
     // console.debug(`Starting ${this.constructor.name}`);
-    if (!this._animationFrameId) {
-      // Wait for start promise before rendering frame
-      this._startPromise = getPageLoadPromise()
-        .then(() => {
-          if (this._stopped) {
-            return null;
-          }
-
-          // Create the WebGL context
-          this._createWebGLContext(opts);
-          this._createFramebuffer();
-          this._startEventHandling();
-
-          // Initialize the callback data
-          this._initializeCallbackData();
-          this._updateCallbackData();
-
-          // Default viewport setup, in case onInitialize wants to render
-          this._resizeCanvasDrawingBuffer();
-          this._resizeViewport();
-
-          // Note: onIntialize can return a promise (in case it needs to load resources)
-          return this.onInitialize(this.animationProps);
-        })
-        .then(appContext => {
-          if (!this._stopped) {
-            this._addCallbackData(appContext || {});
-            if (appContext !== false && !this._animationFrameId) {
-              this._animationFrameId = requestAnimationFrame(this._renderFrame);
-            }
+    // Wait for start promise before rendering frame
+    this._startPromise = getPageLoadPromise()
+      .then(() => {
+        if (!this._running || this._initialized) {
```

Comment:
- Just curious, how can `_running` get set during this time?
- Is `_initialized` the only thing preventing a stopped animation loop from running again? @ibgreen it's not clear from the documentation whether stopping then restarting an animation loop is encouraged, but it was definitely possible before. If we want to keep it that way, there are more issues to be addressed, e.g. every `start` call adds a new canvas.

Reply:
These changes make it so that you can stop/restart an animation loop _without_ running this initialization logic again - that's the second of the bugs listed up top.

Reply:
Edit: read this again, `_initialized` just skips running this block. No breaking change here.